### PR TITLE
Add support for pluto

### PR DESCRIPTION
**Pull request by Dsantra92**
Hi @johnnychen94, I did some work on the Pluto notebook. Sorry, it took so long. Currently, it supports rendering a pluto notebook using PlutoStaticHTML. I am concerned with adding a title and other badges to the rendered notebook. 

However, parsing metadata won't be similar to Julia or markdown's. Pluto.jl has full control over a notebook's start and end content, so we might need to create special `parse` for Pluto that can parse metadata from the file allowing multiline regex matching. I need some advice as to whether I should go ahead with this approach.

### CODE REVIEW

**Comment by johnnychen94** [comment]
@Dsantra92 Great! I'll check this PR this weekend and see if there's anything I can do to quickly get this in.

**Comment by johnnychen94** [comment]
Sorry, it's a busy weekend, I'll manage to get some time for this PR ASAP

**Comment by johnnychen94** [comment]
@Dsantra92 I failed to manage some time doing this (busy work, busy life); really sorry about this!

I didn't check it locally, but the overall changes look okay to me. I just sent you an write invitation to this package so if you're confident, you're free to merge this and add any wanted feature (and potential bug fixes 😆 ).

**Comment by Dsantra92** [comment]
This pr isn't complete yet. I will merge when it's completed.

**Comment by johnnychen94** [comment]
You make the decision

**Comment by Dsantra92** [comment]
I think I can make the docs and basic tests work within the coming weeks. Let's merge this as a beta feature now.

File: src/utils.jl [hunk 1]
```diff
@@ -189,8 +211,14 @@ Currently supported items are: `title`, `id`, `cover`, `description`.
     They also need to validate the values.
 """
 function parse(T::Val, card::AbstractDemoCard)
-    header, frontmatter, body = split_frontmatter(readlines(card.path))
-    config = parse(T, body)
+    # TODO: generalize
+    if T == Val(:Pluto)
```
Comment (CarloLucibello) [inline src/utils.jl:215]:
```suggestion
    if T === Val(:Pluto)
```

**Comment by CarloLucibello** [comment]
Julia v1.0 is not supported by PlutoStaticHTML, we should update the julia version bound of DemoCards.jl to julia v1.6. 

Not clear why windows test on latest julia are failing

File: src/types/pluto.jl [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+* `path`: path to the source markdown file
+* `cover`: path to the cover image
+* `id`: cross-reference id
+* `title`: one-line description of the demo card
+* `author`: author(s) of this demo.
+* `date`: the update date of this demo.
+* `julia`: Julia version compatibility
+* `description`: multi-line description of the demo card
+* `hidden`: whether this card is shown in the generated index page
```
[the remark this answers is not in the document]
Comment (Dsantra92) [reply src/types/pluto.jl:29]: The rest of docs uses small letters, so I let it be.

**Comment by Dsantra92** [comment]
Thanks for the review @CarloLucibello, will get the changes done asap.

**Comment by Dsantra92** [comment]
Hi @CarloLucibello  the changes you suggested to src/types/pluto.jl makes sense but I  then the functions and doc style will differ from the exisiting [markdown ](https://github.com/JuliaDocs/DemoCards.jl/blob/829fd238518243e5372c17b47cdb34d8d9d5b73d/src/types/markdown.jl)and [julia](https://github.com/JuliaDocs/DemoCards.jl/blob/829fd238518243e5372c17b47cdb34d8d9d5b73d/src/types/julia.jl) files.  Just wanted to make these things uniform across the different types.

File: src/types/pluto.jl [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+*This page was generated using [DemoCards.jl](https://github.com/johnnychen94/DemoCards.jl). and [PlutoStaitcHTML.jl](https://github.com/rikhuijzer/PlutoStaticHTML.jl)*
+
+
+"""
+
+"""
+    struct PlutoDemoCard <: AbstractDemoCard
+    PlutoDemoCard(path::String)
+
```
[the remark this answers is not in the document]
Comment (CarloLucibello) [reply src/types/pluto.jl:13]: let's do this

File: src/types/pluto.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (CarloLucibello) [reply src/types/pluto.jl:29]: ok

File: src/types/pluto.jl [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    id::String
+    title::String
+    description::String
+    author::String
+    date::DateTime
+    julia::Union{Nothing, VersionNumber}
+    hidden::Bool
+end
+
+function PlutoDemoCard(path::String)::PlutoDemoCard
```
[the remark this answers is not in the document]
Comment (CarloLucibello) [reply src/types/pluto.jl:73]: let's do this

File: src/types/pluto.jl [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+process the original julia file and save it.
+
+The processing pipeline is:
+
+1. preprocess and copy source file
+3. generate markdown file
+4. insert header and footer to generated markdown file
+"""
+function save_democards(card_dir::String,
```
[the remark this answers is not in the document]
Comment (CarloLucibello) [reply src/types/pluto.jl:108]: let's do this

**Comment by CarloLucibello** [comment]
ok, let's relax to AbstractString and merge and tag quickly then.

**Comment by Dsantra92** [comment]
There should be a minor version bump?

**Comment by CarloLucibello** [comment]
No breaking changes, so bump to 0.4.10

**Comment by Dsantra92** [comment]
I went to bump and release myself. Hope it's okay.

**Pull request merged**